services/cannon/src/Cannon/API.hs

```diff
@@ -170,10 +171,15 @@ singlePush notification (PushTarget usrid conid) = do
         Just x -> do
             e <- wsenv
             b <- notification
-            runWS e $
-                (sendMsg b k x >> return PushStatusOk)
+            (liftIO . timeoutWS sendWsMsgTimeout . runWS e $ sendMsg b k x)
                 `catchAll`
-                const (terminate k x >> return PushStatusGone)
+                const (runWS e (terminate k x) >> return PushStatusGone)
+                -- 'terminate' only clears an MVar in the stream; no need to fork or timeout.
+
+timeoutWS :: Timeout -> IO () -> IO PushStatus
+timeoutWS nanoseconds action = timeout (fromIntegral nanoseconds * 1000) action >>= \case
```

Comment:
Maybe use `t #> Microsecond` here? `fromIntegral` kinda defeats the purpose of all these fancy wrapper types (in fact we probably shouldn't even have that instance 🙈)